### ISSUE TITLE
[GlobalStep] Simple product fails to converse as per the proper conversion when manual renewal of simple subscription on Cart and Checkout page.

### ISSUE THREAD

**Issue by gglobalstep** [Automattic/woocommerce-payments#9458]
### Bug Description:
Simple product fails to converse as per the proper conversion when manual renewal of simple subscription on Cart and Checkout page.

### Environment:
Woocommerce Version: 9.4.0.10
WooCommerce Payments: 8.3.0-test-1
WordPress version: v6.6.1

PC:
Windows 10
Chrome (Version 129.0.6668.59)
Firefox (Version 130.0.1)

### Steps To Reproduce:
1. Create a test site.
2. Install and activate the Woocommerce version 9.4.0.10 plugins.
3. Upload and activate the WooCommerce Payments v8.3.0-test-1 plugin.
4. Complete the setup wizard.
5. Complete the KYC flow.
6. Make sure you have Multi-currency enabled under Payments > Settings > Advanced Settings.
7. Make sure you have a few currencies added, at least one zero decimal currency like JPY or CLP and the multi-currency switcher is enabled under WooCommerce > Settings > Multi-currency.
8. Create a simple subscription product with a signup fee.
9. Purchase your simple subscription product.
10. Navigate to WooCommerce > Subscriptions, and click into one of your new subscriptions.
11. Use Subscription actions at the top right to Create pending renewal order and click the arrow button.
12. Navigate to the new order through the link in the Order notes.
13. The order will have a Customer payment page link above the Status, click that link.
14. Currency switcher widget should be hidden.
15. Confirm that the checkout page has the correct costs listed.
16. Navigate to Shop page and add a simple product to the cart.
17. Go to Cart page.
18. Observe that, Simple product fails to converse as per the proper conversion when manual renewal of simple subscription on Cart and Checkout page.

### Instruction Link
https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-8.3.0#manual-renewal-order-testing

### Actual Result:
Simple product fails to converse as per the proper conversion when manual renewal of simple subscription on Cart and Checkout page.

### Expected Result:
Simple product should converse as per the proper conversion when manual renewal of simple subscription on Cart and Checkout page.

### Video:

https://github.com/user-attachments/assets/d1ea856a-cda6-4012-8642-3851afbb9203

*Isolating the problem (mark completed items with an [x]):*
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

<details>

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.4.0.10
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.1
Log Directory Writable: ✔
WP Version: 6.6.2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.1.29
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 8.7.1
OpenSSL/3.0.11

SUHOSIN Installed: –
MySQL Version: 10.11.9-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 27
page: 10
post: 3
product: 26
product_variation: 9
revision: 6
shop_coupon: 3
shop_order_placehold: 136
wp_font_face: 33
wp_font_family: 12
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (13) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.3
Code Snippets: by Code Snippets Pro – 3.6.5.1
Google for WooCommerce: by WooCommerce – 2.8.5
Jetpack: by Automattic – 13.8.1
Mailchimp for WooCommerce: by Mailchimp – 4.4
Pinterest for WooCommerce: by WooCommerce – 1.4.9
Query Monitor: by John Blackbourn – 3.16.4
TikTok: by TikTok – 1.2.8
WooPayments Dev Tools: by Automattic –
WooPayments: by WooCommerce – 8.3.0-test-1
WooCommerce Shipping & Tax: by WooCommerce – 2.8.1
WooCommerce Subscriptions: by WooCommerce – 6.7.0
WooCommerce: by Automattic – 9.4.0.10

### Inactive Plugins (1) ###

WooCommerce Payments Dev Tools: by Automattic –

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
db.php: Query Monitor Database Class (Drop-in)
object-cache.php: Memcached

### Must Use Plugins (1) ###

load.php: by  –

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 8 MB

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/ -  Contains the woocommerce/cart block
Checkout: #7 - /checkout/ -  Contains the woocommerce/checkout block
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.6.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### WooPayments ###

Version: 8.3.0-test-1
Connected to WPCOM: Yes
WPCOM Blog ID: 237120294
Account ID: acct_1Q0IhfCQzgYGw3Ku
Payment Gateway: Enabled
Test Mode: Enabled
Enabled APMs: card
WooPay: Disabled
WooPay Incompatible Extensions: No
Apple Pay / Google Pay: Enabled (product,cart,checkout)
Fraud Protection Level: basic
Multi-currency: Enabled
Auth and Capture: Enabled
Documents: Disabled
Logging: Enabled

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: 
Subscriptions-core Library Version: 7.5.0
Subscription Statuses: wc-active: 26
wc-cancelled: 2
wc-on-hold: 1

WooCommerce Account Connected: ❌ No
Report Cache Enabled: ✔ Yes
Cache Update Failures: ✔ 0 failure

### Store Setup ###

Country / State: United States (US) — California

### Subscriptions by Payment Gateway ###

WooPayments: wc-active: 26
wc-cancelled: 2
wc-on-hold: 1


### Payment Gateway Support ###

WooPayments: products
refunds
multiple_subscriptions
subscription_cancellation
subscription_payment_method_change_admin
subscription_payment_method_change_customer
subscription_payment_method_change
subscription_reactivation
subscription_suspension
subscriptions
subscription_amount_changes
subscription_date_changes
tokenization
add_payment_method


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings

Daily Cron: ✔ Next scheduled: 2024-09-19 07:20:55 +00:00
Options: ✔
Notes: 62
Onboarding: completed

### Action Scheduler ###

Canceled: 49
Oldest: 2024-09-18 08:09:54 +0000
Newest: 2024-09-18 15:45:57 +0000

Complete: 917
Oldest: 2024-09-18 07:22:06 +0000
Newest: 2024-09-18 09:38:29 +0000

Failed: 3
Oldest: 2024-09-18 07:30:46 +0000
Newest: 2024-09-18 08:46:10 +0000

Pending: 30
Oldest: 2024-09-18 16:10:38 +0000
Newest: 2024-11-18 15:01:29 +0000


### Status report information ###

Generated at: 2024-09-18 16:06:10 +00:00
```
</details>

</details>

**Comment by frosso** [comment]
I assigned this low priority, because it was part of previous releases, and because the issue seems to be reproducible only after visiting the subscription renewal page & adding the product to the cart - I couldn't reproduce it, otherwise.

**Comment by asumaran** [comment]
I was able to reproduce the bug: the amount is not correctly converted on the cart page when the site currency is different from the currency used to purchase the subscription.

**Comment by bborman22** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Fractal-62a0f7b955648b001e1f5e1f/issues/Automattic/woocommerce-payments/9458?planning-poker) with Zenhub @asumaran

**Comment by asumaran** [comment]
In the end, we resolved this not by addressing the currency issue, but by preventing any additional products from being added to the cart when a subscription renewal is being purchased.
